aws/table_aws_shield_attack.go

```diff
@@ -0,0 +1,114 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/shield"
+
+	"github.com/turbot/steampipe-plugin-sdk/v5/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v5/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsShieldAttack(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_shield_attack",
+		Description: "AWS Shield Attack",
+		List: &plugin.ListConfig{
+			Hydrate: listAwsShieldAttack,
+			KeyColumns: plugin.SingleColumn("attack_id"),
+			Tags:    map[string]string{"service": "shield", "action": "DescribeAttack"},
+		},
+		Columns: awsRegionalColumns([]*plugin.Column{
+			{
+				Name:        "attack_id",
+				Description: "The unique identifier (ID) for the attack.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromQual("attack_id"),
+			},
+			{
+				Name:        "resource_arn",
+				Description: "The ARN (Amazon Resource Name) of the resource that was attacked.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ResourceArn"),
+			},
+			{
+				Name:        "sub_resources",
+				Description: "If applicable, additional detail about the resource being attacked, for example, IP address or URL.",
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("SubResources"),
```

Comment:
I'm not quite sure if I understand which change you are suggesting? Do you mean that the `Transform` field is not required since this is done already on a plugin level? Bot how to I define the mapping between table column name and AWS API output field name?

Reply:
For example:

The column `sub_resources` will automatically be populated if the API response includes a property named `SubResources`. This is handled by the default transformation function [FromCamel()](https://github.com/turbot/steampipe-plugin-aws/blob/main/aws/plugin.go#L22), so there’s no need to explicitly add a transform function.

However, for columns like `id`, `url`, or `resource_id`, which are populated from API response attributes like `ID`, `URL`, or `ResourceID`, the default transformation won’t work. In such cases, you’ll need to explicitly use `transform.FromField("ResourceId/ID/URL")` to map them correctly.

Reply:
I'm sorry but I think I did not fully get difference when a transform function is required and when not 😅

What is the difference between a field like `SubResources` and a field like `ResourceID`? Why can the first one directly be transformed to the column name `sub_resources`, while the other one would require a transform function? Is it due to the capitalization of the D in `ResourceID`?

Reply:
> Is it due to the capitalization of the D in `ResourceID`?

Yes, that's correct. 

For more details on transform functions, please refer to the documentation: https://steampipe.io/docs/develop/writing_plugins/transform-functions#transform-functions.

The [plugin-level transform function](https://github.com/turbot/steampipe-plugin-aws/blob/main/aws/plugin.go#L22) is applied globally across all tables in the plugin.

Reply:
Okay, thank you for the explanation. I will adjust the column definitions accordingly and only apply the transform function if needed 👍